src/main/java/org/jenkinsci/plugins/oic/OicSession.java

```diff
@@ -54,7 +55,13 @@
     /**
      * An opaque value used by the client to maintain state between the request and callback.
      */
-    private final String state = Base64.encode(UUID.randomUUID().toString().getBytes(StandardCharsets.UTF_8)).substring(0,20);
+    @VisibleForTesting
+    String state = Base64.encode(UUID.randomUUID().toString().getBytes(StandardCharsets.UTF_8)).substring(0,20);
```

Comment:
Unclear to my this was not just
```suggestion
    String state = UUID.randomUUID().toString();
```
but not touching that here.